### PR TITLE
Psych 4 support

### PR DESCRIPTION
**Pull request by jrafanie**
Part of https://github.com/ManageIQ/manageiq/issues/22696

### CODE REVIEW

File: spec/models/manageiq/providers/openstack/cloud_manager/provision/volume_attachment_spec.rb [hunk 1]
```diff
@@ -5,6 +5,8 @@
     @flavor = FactoryBot.create(:flavor_openstack)
     @volume = FactoryBot.create(:cloud_volume_openstack)
 
+    # We're storing objects in the instance_type, so we must permit loading this class
+    ActiveRecord::Base.yaml_column_permitted_classes = ActiveRecord::Base.yaml_column_permitted_classes | [@flavor.class]
```
Comment (jrafanie) [inline spec/models/manageiq/providers/openstack/cloud_manager/provision/volume_attachment_spec.rb:9]: I'm not a big fan of this but if we use YAML for serialized columns and want to use safe_load, we need to explicitly allow classes to be YAML.loaded at the point we deserialize them... or we have to populate this with all possible models.

File: app/models/manageiq/providers/openstack/helper_methods.rb [hunk 2]
```diff
@@ -64,6 +64,11 @@ def parse_error_message_excon_http_status(exception)
     end
 
     def with_notification(type, options: {})
+      # We're explicitly serializing objects in the options subject key so we should permit yaml loading those classes
```
Comment (agrare) [inline app/models/manageiq/providers/openstack/helper_methods.rb:67]: Wow really?  We should definitely not be doing this, I'll look into this after

File: spec/models/manageiq/providers/openstack/cloud_manager/provision/volume_attachment_spec.rb (hunk 1 above)
Comment (agrare) [reply spec/models/manageiq/providers/openstack/cloud_manager/provision/volume_attachment_spec.rb:9]: Agreed AR instances in MiqRequestTask options is really terrible, but it is everywhere

**Review by agrare — APPROVED** [review]
I want to fix that Notification subject object issue but will do that as a follow-up

File: app/models/manageiq/providers/openstack/helper_methods.rb (hunk 2 above)
Comment (agrare) [reply app/models/manageiq/providers/openstack/helper_methods.rb:67]: https://github.com/ManageIQ/manageiq-providers-openstack/pull/866

**Pull request merged**